### PR TITLE
Markdown package improvements

### LINKED ISSUE #1914: 📘 DOC: Document Markdown rendering
Follow-up to https://github.com/snowpackjs/astro/issues/1349

We still need to document our more pluggable Markdown architecture!

### PR DESCRIPTION
**Pull request by JuanM04**
### `astro`

- Added `MarkdownParser` and `MarkdownParserResponse` to `@types`

### `@astrojs/markdown-remark`

- Upgraded dependencies
- Replaced `remark-slug` with `rehype-slug` because [it was deprecated](https://github.com/remarkjs/remark-slug)
- Replaced `@silvenon/remark-smartypants` with `remark-smartypants` because its name was changed
- Disable **all** built-in plugins when custom remark and/or rehype plugins are added
- Removed `remark-footnotes` because [`remark-gfm` now supports footnotes](https://github.com/remarkjs/remark-gfm/releases/tag/3.0.0)
- Re-added `remark-smartypants` and `rehype-slug` to the default plugins list

## Testing

It wasn't needed to update any tests

## Docs

The page [`/guides/markdown-content`](https://deploy-preview-1954--astro-docs-2.netlify.app/guides/markdown-content/) was updated and fixes #1914

### CODE REVIEW

**Comment by natemoo-re** [comment]
Thanks for tackling this, super helpful! 

Would you mind adding a changeset describing the changes you've made here? 
`yarn changeset` and select `@astrojs/markdown-remark`.

**Comment by JuanM04** [comment]
@natemoo-re I hope I made the changelog right

File: docs/src/pages/guides/markdown-content.md [hunk 1]
```diff
@@ -6,33 +6,63 @@ description: An intro to Markdown with Astro.
 
 Astro comes with out-of-the-box Markdown support powered by the expansive [remark](https://remark.js.org/) ecosystem.
 
-## Remark and Rehype Plugins
+## Parsers
 
-In addition to custom components inside the [`<Markdown>` component](/guides/markdown-content#astros-markdown-component), Astro comes with [GitHub-flavored Markdown](https://github.github.com/gfm/) support, [Footnotes](https://github.com/remarkjs/remark-footnotes) syntax, [Smartypants](https://github.com/silvenon/remark-smartypants), [Remark-slug](https://github.com/remarkjs/remark-slug) and syntax highlighting via [Prism](https://prismjs.com/) pre-enabled.
+Astro let you use any Markdown parser you want. It just need to be a function that follows the `MarkdownParser` type declared inside [this file](https://github.com/snowpackjs/astro/blob/latest/packages/astro/src/@types/astro.ts). You can declare it inside `astro.config.mjs`:
```
Comment (drwpow) [inline docs/src/pages/guides/markdown-content.md:11]: nit: tiny changes

```suggestion
Astro lets you use any Markdown parser you want. It just needs to be a function that follows the `MarkdownParser` type declared inside [this file](https://github.com/snowpackjs/astro/blob/main/packages/astro/src/@types/astro.ts). You can declare it inside `astro.config.mjs`:
```

**Review by drwpow — APPROVED** [review]
LGTM! And good changesets! I‘ll approve but I‘ll give @natemoo-re a chance to give more feedback if desired before merging.

**Review by natemoo-re — APPROVED** [review]
Looks great, thanks so much for the PR!

**Pull request merged**